powerauth-webflow-resources/src/main/java/io/getlime/security/powerauth/lib/webflow/resource/controller/UserProfileController.java

```
@@ -152,7 +151,7 @@ public UserProfileController(DataAdapterClient client, AuthorizationServerTokenS
             logger.info("Found user with ID: {}, given name: {}, family name: {}", usedId, givenName, familyName);
             return new UserInfoResponse(id, id, givenName, familyName, extras);
         } catch (DataAdapterClientErrorException e) {
-            throw new UnauthorizedUserException("Unable to fetch user details from data adapter");
+            throw new OAuth2AuthenticationException("Unable to fetch user details from data adapter");
```

Let's propagate the cause exception or log it at least.